rfd/0098-registered-openssh-nodes.md

```diff
@@ -0,0 +1,103 @@
+---
+authors: Andrew LeFevre (andrew.lefevre@goteleport.com)
+state: draft
+---
+
+# RFD 98 - Registered OpenSSH Nodes
+
+## Required approvers
+
+* Engineering: @jakule && @r0mant
+* Product: @klizhentas
+* Security: @reedloden
+
+## What
+
+Allow OpenSSH nodes to be registered in a Teleport cluster.
+
+## Why
+
+[Agentless EC2 discovery mode](https://github.com/gravitational/teleport/issues/17865) will discover and configure OpenSSH nodes so they can authenticate with a cluster. But those OpenSSH nodes aren't registered as `node` resources in the backend. We need a way to register agentless OpenSSH nodes as `node` resources so they can be viewed and managed by users. RBAC and session recording should function correctly with registered OpenSSH nodes as well.
+
+## Details
+
+### Registering nodes
+
+A new sub-kind to the `node` resource will be added for registered OpenSSH nodes: `openssh`. The absence of a `node` resource sub-kind (or presence of a `teleport` sub-kind) will imply that a node is a Teleport agent node, making this change backwards compatible.
+
+#### Registering with `tctl`
+
+`tctl` should not require as many fields to be set when creating nodes. This is an example `node` resource that will work with `tctl create --force` today:
+
+```yaml
+kind: node
+metadata:
+  name: 5da56852-2adb-4540-a37c-80790203f6a9
+spec:
+  addr: 1.2.3.4:22
+  hostname: agentless-node
+version: v2
+```
+
+`tctl` will auto-generate `metadata.name` if it is not already set so users don't have to generate GUIDs themselves if `sub_kind` is `openssh`. Also, if `sub_kind` is set to `openssh`, `spec.public_addr` will not be allowed for registered OpenSSH nodes as it is not needed. An example of a registered OpenSSH node resource:
+
+```yaml
+kind: node
+sub_kind: openssh
+metadata:
+  name: 5da56852-2adb-4540-a37c-80790203f6a9
+spec:
+  addr: 1.2.3.4:22
+  hostname: agentless-node
+version: v2
+```
+
+### OpenSSH CA
+
+For security related reasons that will be discussed below, a new CA will be added called OpenSSH CA. The OpenSSH CA will be responsible for generating and signing user certificates that will be used to authenticate with registered OpenSSH nodes. The public key of the OpenSSH CA will be copied to all registered OpenSSH nodes and configured as `TrustedUserCAKeys` in `sshd_config`.
+
+#### Manual certificate rotation
+
+Users not using EC2 discovery mode and hosting registered OpenSSH nodes themselves will have to rotate the OpenSSH CA manually. They can do so by:
```

Comment:
We could probably export this CA with minimal additional work to Machine ID for automated rotation - and this has the benefit of being able to issue a host certificate as well.

Reply:
That's a good thought, but I think the Agentless installation/registration code will also be responsible for rotating CAs and host certs. Is that correct @lxea?

Reply:
I believe eventually yes, I'm a bit fuzzy on the details of what else may be supported though.

Reply:
@strideynet We do have the plan to extend the support. First want to add support for AWS and Azure + GCP. For on-premise installations as we don't have any discovery service that we could use to discover all OpenSSH nodes, we will probably require a user to register them manually using `tctl`.

Reply:
👍  nice - just wanted to check if I needed to be in the loop on this but looks like you have it covered.